src/state/hooks.ts

```diff
@@ -78,13 +78,54 @@ export const useFarmUser = (pid) => {
   }
 }
 
-export const useLpTokenPrice = (symbol: string) => {
-  const farm = useFarmFromSymbol(symbol)
-  const tokenPriceInUsd = useGetApiPrice(getAddress(farm.token.address))
+export const useFarmFromTokenSymbol = (tokenSymbol: string, preferredQuoteTokens?: string[]): Farm => {
+  const farms = useSelector((state: State) => state.farms.data.filter((f) => f.token.symbol === tokenSymbol))
+  const filteredFarm = filterFarmsByQuoteToken(farms, preferredQuoteTokens)
+  return filteredFarm
+}
+
+export const useTokenPriceBusd = (pid: number): BigNumber => {
+  const farm = useFarmFromPid(pid)
+  const bnbPriceBusd = usePriceBnbBusd()
+  const quoteTokenFarm = useFarmFromTokenSymbol(farm.token.symbol)
+
+  if (farm.quoteToken.symbol === 'BUSD') {
+    return farm.tokenPriceVsQuote ? new BigNumber(farm.tokenPriceVsQuote) : BIG_ZERO
+  }
+
+  if (farm.quoteToken.symbol === 'wBNB') {
```

Comment:
If the farm has a quoteToken of BUSD or wBNB, it is quite straightforward returning the price.

Reply:
Comment please.